File: Problems/Algorithms/852. Peak Index in a Mountain Array/MountainArray.kt
class Solution {
    fun peakIndexInMountainArray(arr: IntArray): Int {
        var left = 0
        var right = arr.size - 1
        
        while (left < right) {
            var mid = left + (right - left) / 2
            
            if (arr[mid] < arr[mid+1]) {
                left = mid + 1
            } else {
                right = mid
            }
        }
        
        return left
    }
}

File: Problems/Algorithms/852. Peak Index in a Mountain Array/mountain-array.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int peakIndexInMountainArray(vector<int>& arr) {
        int left = -1, right = arr.size();
        while (right - left > 1) {
            int mid = left + (right - left) / 2;
            if (arr[mid] < arr[mid+1]) {
                left = mid;
            } else {
                right = mid;
            }
        }

        return right;
    }
};
